### ISSUE TITLE
Advanced MXP testing

### ISSUE THREAD

**Issue by OlivierBBB** [Consensys/besu-sequencer-plugins#215]
- [x] first step: randomized bytecode that triggers the MXP often without producing errors along the way
  - #197  
  - [ ] TRIVIAL
    - [ ] ROOB
    - [ ] NOOP
  - [ ] NONTRIVIAL_BUT_MXPX
  - [x] NONTRIVIAL
- [x] second step: randomized bytecode that contains meaningful CREATE's
- [ ] ~~third step: same thing but CREATE's finish with nontrivial (randomized) RETURNs: this allows us to deploy nontrivial smart contracts~~ 
- [ ] ~~fourth step: same thing but after the successful deployment remember the deployment address (SSTORE) and later on CALL that address~~
- [ ] fifth step: create a toy world with some predeployed contracts with 
  - bytecode that does a lot of memory operations
  - that calls other contracts in the same toy world (which again do memory operations and maybe do more CALLs)

<img src="https://github.com/Consensys/linea-besu-plugin/assets/38285177/99e22f9b-10ab-4ea3-b9b0-b262cf00320e" width="400">

<img src="https://github.com/Consensys/linea-besu-plugin/assets/38285177/2bfb8b43-659f-4ec3-b736-c5fea6ec9216" width="400">

E.g. Have SMC's A, B, C, D of the above type and craft a tx with:
- Tx.to = A.address
- A does memory operations
- A calls B
  - B does memory operations
  - B returns to A
- A does memory operations
- A calls C
  - C does memory operations
  - C calls B
    - B does memory operations
    - B returns to C
  - C does memory operations
  - C calls D 
    - D does memory operations
    - D returns to C
  - C does memory operations
  - C returns to A
- A does memory operations
- A returns and finishes the tx

# Meaningful CREATE's

- produce a function `nontrivialCreate()` that returns a _slice of bytes_ INIT that will be the initialization code
	- use `appendOpcodeCall()` many times to fill INIT, finishing on a RETURN or REVERT
- produce a function called `randomBytecodeWithCreates()` that will return a random bytecode
	- appending random opcode calls like previously
	- from time to time invoke the `nontrivialCreate()` function to produce a slice of bytes INIT
		- choose a random small offset called INITCODEOFFSET
		- take that slice of bytes INIT, chop it up into 32B chunks
		- PUSH32 a chunk, and MSTORE it starting INITCODEOFFSET
		- every new chunk is again PUSH32'd, MSTORE'd at the the previous offset + 32
		- when the code has been stored in memory in its entirety we invoke `appendOpcodeCall()` with the following instruction:
			- OpCode  = CREATE (or CREATE2 but add some random 32B salt to it) 
			- offset1 = INITCODEOFFSET
			- size1   = |INIT| (length of the slice of bytes)
			- value   = some random small number (< 10**18)
	- rinse and repeat
		- i.e. to this several times within a single global bytecode that you produce

This will allow you to test for very cheap the coherence conditions

# Producing meaningful deployments

You can then: also tests CALL's quite cheaply if you first do something like this:
- the final RETURN instruction that you produced above should, too, have interesting arguments
- you can create, similiarly as above, a function called `nontrivialReturn()` that:
  - produces some interesting DEPLOYMENTCODE (some bytecode produced like INIT earlier)
  - sticks this DEPLOYMENTCODE into memory (of the CREATEE) with again some PUSH32's and MSTORE's
  - at the end returns that slice of bytecode
- then you can after the CREATE(2) instruction you have the address on the stack
- you can then use that address (maybe SSTORE it)
- and use this address as an argument for CALL's happening later

This requires some care:
- the `nontrivialReturn()` must be invoked within the `nontrivialCreate()` funtion
- in other words, we should append such a `nontrivialReturn()` to the end of the INIT code we use.

# Quick way to test RETURNDATACOPY

One quick way to test this opcode without deploying contracts first which produce returndata is to use CALL's to precompiles, typically:
- SHA2-256: return data measures 32 bytes
- RIPEMD-160: return data measures 32 bytes
- IDENTITY: return data measures size1 bytes

# Doing CALL's quickly

We can use the example of [this toyworld](https://github.com/Consensys/linea-besu-plugin/blob/ba37963098e0b4decfc083cdbd13fbe05992cae8/src/test/java/net/consensys/linea/zktracer/testing/ExampleTxTest.java) to deploy bytecode without having to do the CREATE process ourselves. With this in hand it will be easy for us to test CALL's and RETURNDATACOPY's.
